Source/Core/Core/PowerPC/JitCommon/JitCache.h

```diff
@@ -84,28 +111,26 @@ class JitBaseBlockCache
     MAX_NUM_BLOCKS = 65536 * 2,
   };
 
-  std::array<const u8*, MAX_NUM_BLOCKS> blockCodePointers;
   std::array<JitBlock, MAX_NUM_BLOCKS> blocks;
   int num_blocks;
   std::multimap<u32, int> links_to;
   std::map<std::pair<u32, u32>, u32> block_map;  // (end_addr, start_addr) -> number
+  std::map<u32, u32> start_block_map;            // start_addr -> number
   ValidBlockBitSet valid_block;
 
-  bool m_initialized;
-
   void LinkBlockExits(int i);
   void LinkBlock(int i);
   void UnlinkBlock(int i);
 
-  u8* GetICachePtr(u32 addr);
   void DestroyBlock(int block_num, bool invalidate);
 
   // Virtual for overloaded
   virtual void WriteLinkBlock(u8* location, const JitBlock& block) = 0;
   virtual void WriteDestroyBlock(const u8* location, u32 address) = 0;
+  virtual void WriteUndestroyBlock(const u8* location, u32 address) = 0;
```

Comment:
I think your assessment of the situation is correct... but I'm not sure messing with BLR optimization as opposed to the checked entry of a function is the right tradeoff.